script/homm2/extract_homm2_resources_from_cd_image.sh

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env bash
+
+###########################################################################
+#   fheroes2: https://github.com/ihhub/fheroes2                           #
+#   Copyright (C) 2025                                                    #
+#                                                                         #
+#   This program is free software; you can redistribute it and/or modify  #
+#   it under the terms of the GNU General Public License as published by  #
+#   the Free Software Foundation; either version 2 of the License, or     #
+#   (at your option) any later version.                                   #
+#                                                                         #
+#   This program is distributed in the hope that it will be useful,       #
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of        #
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         #
+#   GNU General Public License for more details.                          #
+#                                                                         #
+#   You should have received a copy of the GNU General Public License     #
+#   along with this program; if not, write to the                         #
+#   Free Software Foundation, Inc.,                                       #
+#   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             #
+###########################################################################
+
+set -e -o pipefail
+
+function echo_red {
+    echo -e "\033[0;31m$*\033[0m"
+}
+
+function echo_green {
+    echo -e "\033[0;32m$*\033[0m"
+}
+
+function echo_stage {
+    echo
+    echo_green "$*"
+    echo
+}
+
+echo_green "This script will extract and copy game resources from the original Heroes of Might and Magic II CD images"
+
+echo_stage "[1/4] determining the destination directory"
+
+DEST_PATH=""
+
+if [[ -n "$2" ]]; then
+    DEST_PATH="$2"
+elif [[ -f fheroes2 && -x fheroes2 ]]; then
+    DEST_PATH="."
+elif [[ -d ../../src ]]; then
+    # Special hack for developers running this script from the source tree
+    DEST_PATH="../.."
+fi
+
+if [[ -z "$DEST_PATH" || ( -d "$DEST_PATH" && ! -w "$DEST_PATH" ) ]]; then
+    if [[ "$(uname 2> /dev/null)" == "Linux" ]]; then
+        DEST_PATH="${XDG_CONFIG_HOME:-$HOME/.local/share}/fheroes2"
+    elif [[ -z "$2" ]]; then
+        DEST_PATH="$HOME/.fheroes2"
+    fi
+fi
+
+echo_green "Destination directory: $DEST_PATH"
+
+echo_stage "[2/4] determining the path to the HoMM2 CD image"
+
+if [[ "$#" -gt "0" ]]; then
+    HOMM2_CD_IMAGE_PATH="$1"
+else
+    read "HOMM2_CD_IMAGE_PATH?Please enter the full path to the HoMM2 cd image (e.g. /home/user/homm2.cue or /home/user/homm2.iso): "
```

Comment:
This syntax of the `read` command is valid for the `zsh`, but not for `bash`. I suggest to use the following syntax for `bash`:

```suggestion
    read -e -p "Please enter the full path to the HoMM2 cd image (e.g. /home/user/homm2.cue or /home/user/homm2.iso): " HOMM2_CD_IMAGE_PATH
```

`-e` allows `bash` to use the `readline` library to perform the file name completion using TAB character, which is handy when you need to enter the name of an existing file.

Reply:
Nice catch!  I'll fix this.